Flaky connection closed test

**Introduction**
The recently added test appears flaky on Jenkins. @sondreso points out that it is probably due to the test ensemble not being cancellable and hence it is stopped. Then we cannot guarantee that the asserted number of realisations will be successful and the assert fails.

**Solution**
A possible solution would be to make either the `TestEnsemble` or `AutorunTestEnsemble` cancellable by just not doing anything when cancelled and hence finishing its thing. The upside of doing it only for `AutorunTestEnsemble` is that there are no other tests depending on it and hence the cascading effect should be none. The upside of doing it for `TestEnsemble` is that we could stop having logic for whether it is cancellable or not in the code...

**Stacktrace**
```
=================================== FAILURES ===================================
_ test_ensemble_evaluator_run_and_get_successful_realizations_connection_closed_recover[10-5] _

make_ee_config = <function make_ee_config.<locals>._ee_config at 0x7f67151ceb70>
num_realizations = 10, num_failing = 5

    @pytest.mark.parametrize("num_realizations, num_failing", [(10, 5), (10, 10)])
    def test_ensemble_evaluator_run_and_get_successful_realizations_connection_closed_recover(
        make_ee_config, num_realizations, num_failing
    ):
        ee_config = make_ee_config(use_token=False, generate_cert=False)
        ensemble = AutorunTestEnsemble(iter=1, reals=num_realizations, steps=1, jobs=2)
        for i in range(num_failing):
            ensemble.addFailJob(real=i, step=0, job=1)
        ee = EnsembleEvaluator(ensemble, ee_config, 0, ee_id="0")
        with patch.object(
            _Monitor,
            "track",
            side_effect=[get_connection_closed_excpetion()] * 2
            + [ConnectionRefusedError("Connection error")]
            + [dummy_iterator("DUMMY TRACKING ITERATOR")]
            + [get_connection_closed_excpetion()] * 2
            + [ConnectionRefusedError("Connection error")] * 2
            + ["DUMMY TRACKING ITERATOR2"],
        ) as mock:
            num_successfull = ee.run_and_get_successful_realizations()
            assert mock.call_count == 9
>       assert num_successfull == num_realizations - num_failing
E       assert 0 == (10 - 5)

ensemble_evaluator/test_ensemble_evaluator.py:312: AssertionError
=================================== FAILURES ===================================
_ test_ensemble_evaluator_run_and_get_successful_realizations_connection_closed_recover[10-5] _

make_ee_config = <function make_ee_config.<locals>._ee_config at 0x7f67151ceb70>
num_realizations = 10, num_failing = 5

    @pytest.mark.parametrize("num_realizations, num_failing", [(10, 5), (10, 10)])
    def test_ensemble_evaluator_run_and_get_successful_realizations_connection_closed_recover(
        make_ee_config, num_realizations, num_failing
    ):
        ee_config = make_ee_config(use_token=False, generate_cert=False)
        ensemble = AutorunTestEnsemble(iter=1, reals=num_realizations, steps=1, jobs=2)
        for i in range(num_failing):
            ensemble.addFailJob(real=i, step=0, job=1)
        ee = EnsembleEvaluator(ensemble, ee_config, 0, ee_id="0")
        with patch.object(
            _Monitor,
            "track",
            side_effect=[get_connection_closed_excpetion()] * 2
            + [ConnectionRefusedError("Connection error")]
            + [dummy_iterator("DUMMY TRACKING ITERATOR")]
            + [get_connection_closed_excpetion()] * 2
            + [ConnectionRefusedError("Connection error")] * 2
            + ["DUMMY TRACKING ITERATOR2"],
        ) as mock:
            num_successfull = ee.run_and_get_successful_realizations()
            assert mock.call_count == 9
>       assert num_successfull == num_realizations - num_failing
E       assert 0 == (10 - 5)

ensemble_evaluator/test_ensemble_evaluator.py:312: AssertionError
```
**Full stacktrace**
https://ci.equinor.com/scout/job/komodo_testing/job/ert-custom-script-pr/6153/console

--- a/tests/ert_tests/ensemble_evaluator/ensemble_evaluator_utils.py
+++ b/tests/ert_tests/ensemble_evaluator/ensemble_evaluator_utils.py
@@ -222,3 +222,9 @@ class AutorunTestEnsemble(TestEnsemble):
         )
 
         self._eval_thread.start()
+
+    def cancel(self):
+        pass
+
+    def is_cancellable(self):
+        return True

